### PR TITLE
[Test] Use TMPDIR if available

### PR DESCRIPTION
**Pull request by arkamar**
Couple of tests were designed to use hardcoded `/tmp` directory, which could be problematic in some cases. For example, in case of sqlite3 related tests, multiple users cannot run it in parallel. Additionally some leftovers remain in `/tmp` directory after this test, `*.sqlite-shm` and `*.sqlite-wal` files, which was actually main motivation for me to create this PR.

From perspective of Gentoo packager `$TMPDIR` is set for the package to a safe place controlled by a package manager, which is automatically cleaned when something remains there. Moreover, I can run tests for multiple versions in parallel without risk of file collision.

### CODE REVIEW

File: src/libutil/cxx/file_util.cxx [hunk 1]
```diff
@@ -368,13 +368,22 @@ TEST_CASE("check lock") {
...
 }
 
+const char * get_tmpdir() {
+	const char *tmpdir = getenv("TMPDIR");
+	if (tmpdir == nullptr) {
+		tmpdir = G_DIR_SEPARATOR_S "tmp";
+	}
+	return tmpdir;
+}
+const std::string tmpdir{get_tmpdir()};
```
Comment (vstakhov) [inline src/libutil/cxx/file_util.cxx:378]:
```suggestion
static const std::string tmpdir{get_tmpdir()};
```

Comment (vstakhov) [reply src/libutil/cxx/file_util.cxx:378]: Or probably it should be a `TEST_CASE` local variable - even better.

File: src/libutil/cxx/file_util.cxx [hunk 2]
```diff
@@ -368,13 +368,22 @@ TEST_CASE("check lock") {
...
+
 TEST_CASE("tempfile") {
 	std::string tmpname;
 	{
-		auto raii_locked_file = raii_locked_file::mkstemp("/tmp//doctest-XXXXXXXX",
+		auto raii_locked_file = raii_locked_file::mkstemp(std::string(tmpdir + "//doctest-XXXXXXXX").c_str(),
```
Comment (vstakhov) [inline src/libutil/cxx/file_util.cxx:383]:
```suggestion
		auto raii_locked_file = raii_locked_file::mkstemp(std::string(tmpdir + G_DIR_SEPARATOR_S + "doctest-XXXXXXXX").c_str(),
```

File: src/libutil/cxx/file_util.cxx [hunk 3]
```diff
@@ -391,10 +400,10 @@ TEST_CASE("tempfile") {
 TEST_CASE("mmap") {
 	std::string tmpname;
 	{
-		auto raii_file = raii_file::mkstemp("/tmp//doctest-XXXXXXXX",
+		auto raii_file = raii_file::mkstemp(std::string(tmpdir + "//doctest-XXXXXXXX").c_str(),
```
Comment (vstakhov) [inline src/libutil/cxx/file_util.cxx:403]:
```suggestion
		auto raii_file = raii_file::mkstemp(std::string(tmpdir + G_DIR_SEPARATOR_S + "doctest-XXXXXXXX").c_str(),
```

**Review by vstakhov — APPROVED** [review]
The suggested changes look good in general, however, I'd suggest some minor neats.

**Comment by arkamar** [comment]
I changed it according to your suggestions.

**Comment by arkamar** [comment]
Those failing tests in CI do not seem to be related to my changes:
```
==============================================================================
Lua Check                                                             | FAIL |
Checking src/plugins/lua/antivirus.lua            1 warning

    src/plugins/lua/antivirus.lua:151:1: (W611) line contains only whitespace

Total: 1 warning / 0 errors in 183 files
------------------------------------------------------------------------------
Cases.200 Luacheck                                                    | FAIL |
1 test, 0 passed, 1 failed
==============================================================================
```

**Pull request merged**